framework/include/cppmicroservices/BundleContext.h

```diff
@@ -668,11 +826,11 @@ class US_Framework_EXPORT BundleContext
    * @see RemoveServiceListener()
    */
   template<class R>
-  void AddServiceListener(R* receiver, void(R::*callback)(const ServiceEvent&),
-                          const std::string& filter = std::string())
+  ListenerToken AddServiceListener(R* receiver, void(R::*callback)(const ServiceEvent&),
```

Comment:
Tagging @jeffdiclemente @karthikreddy09 @saschazelzer 

I am having second thoughts about deprecating these three Add.*Listener functions (that take member functions). Reasons:

1. Though the functionality can be replicated by the other three Add.*Listener functions, these three seem to be convenient wrappers for adding member functions.
2. Internally, I see the usage of these functions to be more prevalent. If deprecated, all the invocations have to be changed to std::bind form, that maybe inconvenient to clients i.e. If these are deprecated, the negatives outweigh the benefits. Do you feel about this strongly, Sascha and others?